### PR TITLE
Fix Cloud Build Issue

### PR DESCRIPTION
**Pull request by Nightsd01**
• In previous versions of the SDK, developers had to make manual changes to their Unity generated iOS Xcode projects (like adding UserNotifications.framework) in order to build.
• This PR makes some minor changes to the SDK to automatically add UserNotifications.framework and changes some build settings so projects can build without errors
• Also fixes an issue that would have made iOS unable to compile due to an extraneous semicolon in OneSignalIOS.cs

### CODE REVIEW

File: OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+         foreach (framework framework in frameworksToAdd) {
+            AddFrameworkToProject(framework, ref contents);
+         }
+
+         File.WriteAllText(projectFile, contents);
+      }
+   }
+
+   public static void AddFrameworkToProject(framework framework, ref string contents) {
+      contents = contents.Replace("Ref = 8AC71EC319E7FBA90027502F /* OrientationSupport.mm */; };", "Ref = 8AC71EC319E7FBA90027502F /* OrientationSupport.mm */; };\n\t\t" + framework.id + " /* " + framework.name + ".framework in Frameworks */ = {isa = PBXBuildFile; fileRef = " + framework.fileId + " /* " + framework.name + ".framework */; };");
```
Comment (jkasten2) [inline OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs]: Not repeating each find would make it easier to see the changes being made by the build script.

File: OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_OneSignal.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+﻿using UnityEngine;
```
[the remark this answers is not in the document]
Comment (Nightsd01) [reply OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_OneSignal.cs]: Old Commit - file removed

File: OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+public class PostBuildTrigger {
+
+	[PostProcessBuild]
+	public static void OnPostProcessBuild(BuildTarget target, string pathToBuiltProject) {
+		LinkLibraries(target, pathToBuiltProject);
+	}
+
+	public static void LinkLibraries(BuildTarget target, string pathToBuiltProject) {
+		if(target == BuildTarget.iOS)
```
[the remark this answers is not in the document]
Comment (Nightsd01) [reply OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs]: Fixed - broke strings into their own variables

File: OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs (hunk 1 above)
Comment (Nightsd01) [reply OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs]: Fixed - broke strings into their own variables

File: OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nightsd01) [reply OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs]: Removed

**Pull request merged**